src/messaging/ReliableMessageManager.cpp

```diff
@@ -37,30 +37,69 @@ namespace chip {
 namespace Messaging {
 
 ReliableMessageManager::RetransTableEntry::RetransTableEntry() :
-    rc(nullptr), msgBuf(nullptr), msgId(0), msgSendFlags(0), nextRetransTimeTick(0), sendCount(0)
+    rc(nullptr), msgBuf(nullptr), peerNodeId(0), msgId(0), nextRetransTimeTick(0), sendCount(0)
 {}
 
 ReliableMessageManager::ReliableMessageManager() :
-    mTimeStampBase(System::Timer::GetCurrentEpoch()), mCurrentTimerExpiry(0),
+    mSystemLayer(nullptr), mContextPool(nullptr), mCurrentTimerExpiry(0),
     mTimerIntervalShift(CHIP_CONFIG_RMP_TIMER_DEFAULT_PERIOD_SHIFT)
 {}
 
 ReliableMessageManager::~ReliableMessageManager() {}
 
-void ReliableMessageManager::ProcessDelayedDeliveryMessage(ReliableMessageContext * rc, uint32_t PauseTimeMillis)
+void ReliableMessageManager::Init(chip::System::Layer * systemLayer, SecureSessionMgr * sessionMgr, ExchangeContext * contextPool)
+{
+    mSystemLayer = systemLayer;
+    mSessionMgr  = sessionMgr;
+    mContextPool = contextPool;
+
+    memset(RetransTable, 0, sizeof(RetransTable));
```

Comment:
for loop setting everything to nullptr may be more readable.

Retranstable should be mRetransmissionTable

Reply:
Changed to mRetransTable. The context of this array is not pointer,but struct RetransTableEntry.

we can implement reset construct of struct RetransTableEntry to avoid memset, but we still need to memset PayloadHeader within RetransTableEntry